src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const a of actions) {
+    switch (a.type) {
+      case 'addProperties':
+        Object.assign(state, a.extraData);
+
+        break;
+
+      case 'removeProperties':
+        const takeAway = a.keysToRemove;
+
+        for (const name of takeAway) {
+          if (state[name]) {
```

Comment:
not the best way, because property may have falsy value and you don't remove it. it's better to use `hasOwnProperty`

Reply:
what do you mean by falsy value?